2021-10-13-github-templates.md

```diff
@@ -0,0 +1,123 @@
+# OmbuLabs Standard Github Templates
+
+## Problem:
+
+New and junior contributors might be unsure about the information we require to
+efficiently review a PR. Similarly, it might be unclear what information is required when
+a new BUG or FEATURE is logged as an issue in Github.
+
+We have existing PR and ISSUE templates but they are inconsistent and might ask for
+unnecessary information from the contributor.
+
+The goal is to simplify  and streamline the steps required to make a contribution or log
+an issue.
+
+## Proposal:
+
+Come up with a consistent set of Github templates. The templates should include prompts
+only the ESSENTIAL information we require. We want to make sure that we don't ask for more
+information than we require - as this will discourages contributions.
+
+## Templates:
+
+### Pull Requests
+
+```
+    <!--- Before we start... -->
+    <!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+    <!--- Did you update any relevant documentation -->
+    <!--- Did you add tests to cover new changes -->
```

Comment:
@fbuys hm, what's the point of these two questions? The reviewer can easily have an answer by looking at the files changed.

If it's just a reminder for the person creating the PR maybe we should rephrase them to something like:
- Make sure to update any relevant documentation
- Make sure to add automated tests for the new changes

Reply:
I think we could have a `how to contribute` document that explains all the details, having this in the template just adds extra work removing them.

Also reviewers should note that tests or documentation is missing to ask for that.

Reply:
https://docs.github.com/en/enterprise-server@3.2/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#about-contributing-guidelines github supports a CONTRIBUTING file that pops up this message for new contributors

Reply:
though I think this should be better discussed with the open source team, I think it's a good idea to leverage that github feature to cleanup the templates, but then we need a good `contributing.md` file that we can reuse in projects

Reply:
i'm into the idea of adding `contributing.md` files, I usually find them really helpful.